resources/js/components/follow-toggle.tsx

```diff
@@ -3,58 +3,50 @@
 
 import FollowJson from 'interfaces/follow-json';
 import { route } from 'laroute';
+import { action, makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
 import core from 'osu-core-singleton';
 import * as React from 'react';
 import { classWithModifiers, Modifiers } from 'utils/css';
-import { nextVal } from 'utils/seq';
+import { parseJsonNullable, storeJson } from 'utils/json';
 import { Spinner } from './spinner';
 
 interface Props {
   follow: FollowJson;
-  following: boolean;
   modifiers?: Modifiers;
 }
 
-interface State {
+interface SavedState {
   following: boolean;
-  toggling: boolean;
 }
 
-// TODO: mobx and turn to observer
-export default class FollowToggle extends React.PureComponent<Props, State> {
-  static defaultProps = {
-    following: true,
-  };
-
-  state: State;
-
-  private readonly eventId = `follow-toggle-${nextVal()}`;
-  private toggleXhr: null | JQueryXHR = null;
+@observer
+export default class FollowToggle extends React.PureComponent<Props> {
+  @observable private _following = true;
+  private readonly jsonId = `json-follow-toggle-${this.props.follow.subtype}-${this.props.follow.notifiable_type}-${this.props.follow.notifiable_id}`;
```

Comment:
this should be a getter? doesn't matter now but it'll probably break if the component is ever shuffled around and get passed a different set of props